### DISCUSSION TITLE
Is there any advice about how to migrate the Django Q backend from AWS SQS to Redis?

### DISCUSSION THREAD

**Discussion by rukbotto** [django-q2/django-q2#226] [Q&A] [answered]
I'm working in a Django project using `django-q2` configured to work with AWS SQS. However, due to performance issues we are recently having, we want to change the backend to Redis. Wondering if there's some advice or best practices about how to do this in a working production application.

The [configuration change](https://django-q2.readthedocs.io/en/master/configure.html#redis) to make it work with Redis seems to be the easy part. I'm just thinking about how to deal with pending tasks left on SQS at the time of switching backends, or additional considerations that we are not aware of.

Thanks in advance!

**Comment by GDay** [comment, marked as answer]
This is going to be very tricky and is not supported out of the box. You will have to do some trial and error on a staging environment to get this right. 

Can you afford some downtime? If so, that would make things a lot easier. Stop all schedules and disable task creation (shut down the site? but leave the background worker running) and just let all pending tasks finish and then move over to Redis.

If you can only afford a little bit of downtime, then you will have to pull the data from sqs and then push it up to Redis. The data should be the same, as django-q(2) is pretty much broker agnostic - there are no special things between the payload that is getting send to the broker. You could write a migration script for this. Run this right after django-q(2) has shut down. Pull all messages from SQS and then push them to Redis. Then restart the app with using Redis. 

If you cannot afford any downtime at all (assuming you are running multiple instances of django-q2 at any time), you would just have to leave one cluster running that is connected to SQS until all tasks are done. Spin up a new cluster connected to Redis which would start running the newly scheduled tasks. Depending on how you deploy stuff, this could be pretty tricky to setup on the devops/infra side.

In theory, these options should all work, but I have never had to do this, so don't quote me on it.

**Reply by rukbotto** [reply]
Really helpful insights, thanks a lot!
